### DISCUSSION TITLE
Working with state - custom state class

### DISCUSSION THREAD

**Discussion by rabol** [spatie/laravel-livewire-wizard#116] [Q&A]
Hi

I have a need to skip / remove some steps in a wizard, and I would like to store some default values for the information that would normally be retrieved in the step(s) that is removed.

The way that I handle which steps that I will show / hide is like this:

```
    private array $customSteps = [];

    public function steps(): array
    {
        return $this->customSteps;
    }
```

then in the mount() method I fill the ```$this->customSteps```, that seems to be working.

I cannot figure how I can set values on my state class in the mount() method of the wizard component and then use the information in one or more steps.

If I understand the code correctly, a State object is only created when on try to access the ```$this->state()``` in a step, or is that wrong ? 

So, if the above is correct doe that mean that you cannot access the state object in the mount method of the WizardComponent?

**Comment by rabol** [comment]
What I have come up with so far is:

```
public function setStepState(string $step, array $state = []): void
    {
        if ($step !== 'global') {
            if (! $this->stepNames()->contains($step)) {
                throw StepDoesNotExist::doesNotHaveState($step);
            }
        }

        $this->allStepState[$step] = $state;
    }
```

Then in each step I can do ```$this->state()->forStep('global')``` but it does not feel 100% correct

**Comment by ju5t** [comment]
This should do it: https://spatie.be/docs/laravel-livewire-wizard/v2/usage/setting-initial-state.

```php
public function initialState(): array
{
    $user = User::findOrFail($this->userId);
    
    return [
        'delivery-address-step' => [
            'zip' => $user->zip,
            'city' => $user->city,
        ],       
    ];
}
```

**Reply by rabol** [reply]
this only works if you have a step called 'delivery-address-step', I would like to store some 'global' values

**Reply by ju5t** [reply]
You can use `initialState()` to make sure state is available in the steps that need it. That's the 'supported' way. If that's not what you're after your solution works too. It's not very likely that the `setStepState` method will change in the future but it's something to be aware of.

**Reply by rabol** [reply]
ok, got your point - i need to add my custom data to a step and not have global data
